lib/server/controllers/qcFlagTypes.controller.js

```diff
@@ -37,6 +40,43 @@ const getQcFlagTypeByIdHandler = async (req, res) => {
     }
 };
 
+// eslint-disable-next-line valid-jsdoc
+/**
+ * List All QCFlagReasons
+ */
+const listQCFlagTypesHandler = async (req, res) => {
+    const validatedDTO = await dtoValidator(
+        DtoFactory.queryOnly({
+            filter: {
+                ids: Joi.array().items(Joi.number()),
+                bad: Joi.boolean(),
+                archived: Joi.boolean(),
+                names: Joi.alternatives().try(Joi.array().items(Joi.string()), Joi.object({ like: Joi.array().items(Joi.string()) })),
+                methods: Joi.alternatives().try(Joi.array().items(Joi.string()), Joi.object({ like: Joi.array().items(Joi.string()) })),
+            },
+            page: PaginationDto,
+            sort: DtoFactory.order(['id', 'name', 'bad', 'method', 'createdAt', 'updatedAt']),
```

Comment:
Maybe archived too, no? Isn't it one of the default? 
For tags I think it's `archived` then `text`, I think we should do the same

Reply:
applied